sim/paladin/holy_power_bar.go

```diff
@@ -0,0 +1,60 @@
+package paladin
+
+import "github.com/wowsims/mop/sim/core"
+
+type HolyPowerBar struct {
+	resourceBar core.SecondaryResourceBar
+	paladin     *Paladin
+}
+
+// RegisterOnGain implements core.SecondaryResourceBar.
+func (h HolyPowerBar) RegisterOnGain(callback core.OnGainCallback) {
+	h.resourceBar.RegisterOnGain(callback)
+}
+
+// RegisterOnSpend implements core.SecondaryResourceBar.
+func (h HolyPowerBar) RegisterOnSpend(callback core.OnSpendCallback) {
+	h.resourceBar.RegisterOnSpend(callback)
+}
+
+// CanSpend implements core.SecondaryResourceBar.
+func (h HolyPowerBar) CanSpend(limit int32) bool {
+	return h.Value() >= limit
+}
+
+// Gain implements core.SecondaryResourceBar.
+func (h HolyPowerBar) Gain(amount int32, action core.ActionID, sim *core.Simulation) {
+	h.resourceBar.Gain(amount, action, sim)
+}
+
+// Reset implements core.SecondaryResourceBar.
+func (h HolyPowerBar) Reset(sim *core.Simulation) {
+	h.resourceBar.Reset(sim)
+}
```

Comment:
You can delete all this shared boilerplate if you directly embed `DefaultSecondaryResourceBarImpl` in the struct:
```
type HolyPowerBar struct {
	*core.DefaultSecondaryResourceBarImpl
	paladin *Paladin
}
```
This guarantees that `HolyPowerBar` implements `core.SecondaryResourceBar`, so you only need to supply your own implementations of the few interface methods that differ from the default (`Spend()`, `SpendUpTo()`, and `Value()`).